lib/active_record/connection_adapters/sqlserver/jdbc.rb

```diff
@@ -0,0 +1,49 @@
+begin
+  eval 'com.microsoft.sqlserver.jdbc.SQLServerDriver'
+rescue NameError
+  begin
+    require 'sqljdbc4'
+  rescue
+    raise("com.microsoft.sqlserver.jdbc.SQLServerDriver not loaded, try installing the sqljdbc4 gem or put the jdbc driver jar in the java CLASSPATH")
```

Comment:
Is there a public `sqljdbc4` gem? I can't find it on rubygems.

Reply:
There's one on rubygems from 2014 and then @iaddict has one avail via github but not published back to RubyGems from 2016.

- https://rubygems.org/gems/jdbc-sqlserver/versions/0.0.2
- https://github.com/iaddict/sqljdbc4-java

The jar has been updated a fair bit since then. The

Reply:
There's a few (too many) choices for your gem wrapper

- https://rubygems.org/gems/do-jdbc_sqlserver
- https://rubygems.org/gems/jdbc-sqlserver
- https://rubygems.org/gems/jdbc-mssqlserver
- https://github.com/iaddict/sqljdbc4-java
- https://rubygems.org/gems/jdbc-mssql

The most recent one is jdbc-mssql. I'd say go with that one.

I haven't checked into which driver one would want to use but there's 4.0, 4.1, 4.2, 6.x, 7.x versions that are published by MS. I wonder if there's reasons not to use higher versioned drivers.  Should we expect to use the v6 or v7 driver?

Reply:
seems jdbc-mssql will need some work, [will "only" work on Java 7/8 atm](https://github.com/JesseChavez/jdbc-mssql/blob/master/lib/jdbc/mssql.rb#L14) ... the reason why AR-JDBC hasn't packaged anything exceptopen-source drivers is licensing, which I am not sure about.

Reply:
actually, .jars are available on Maven these days, and their declared as MIT, so all good
... https://search.maven.org/artifact/com.microsoft.sqlserver/mssql-jdbc/6.2.2.jre8/jar